### PR TITLE
feat: add tf-summarize

### PR DESCRIPTION
**Pull request by CalvinRodo**
# Summary | Résumé

This PR does multiple things

- Update the docker file with gh cli, and ssh forwarding in
   devcontainer
- add tf-summarize to devcontainer
- add tf-summarize to pr-test
- add a summary step that will print out the summary of changes

This is a breaking change to this script since it requires tf-summarize to be installed locally in the build

### CODE REVIEW

File: test.json [hunk 1]
```diff
@@ -0,0 +1 @@
...es":{"outputs":{"id":{"sensitive":false}},"root_module":{"resources":[{"address":"random_id.id","mode":"managed","type":"random_id","name":"id","provider_name":"registry.terraform.io/hashicorp/random","schema_version":0,"values":{"byte_length":8,"keepers":null,"prefix":null},"sensitive_values":{}}]}},"resource_changes":[{"address":"random_id.id","mode":"managed","type":"random_id","name":"id","provider_name":"registry.terraform.io/hashicorp/random","change":{"actions":["create"],"before":null,"after":{"byte_length":8,"keepers":null,"prefix":null},"after_unknown":{"b64_std":true,"b64_url":true,"dec":true,"hex":true,"id":true},"before_sensitive":false,"after_sensitive":{}}}],"output_changes":{"id":{"actions":["create"],"before":null,"after_unknown":true,"before_sensitive":false,"after_sensitive":false}},"configuration":{"provider_config":{"test":{"name":"test","version_constraint":"~\u003e 3.1.0"}},"root_module":{"outputs":{"id":{"expression":{"references":["random_id.id.hex","random_id.id"]}}},"resources":[{"address":"random_id.id","mode":"managed","type":"random_id","name":"id","provider_config_key":"random","expressions":{"byte_length":{"constant_value":8}},"schema_version":0}]}}}
```
Comment (patheard) [inline test.json:1]: nit: can you delete this test file

Comment (CalvinRodo) [reply test.json]: done

**Review by gcharest — COMMENTED** [review]
I'm running into an issue with SSH config. Trying to isolate source as it might be local.

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -9,11 +9,17 @@
...
             "version": "latest",
             "tflint": "none",
-            "terragrunt": "none"
-        }
+            "terragrunt": "latest"
+        },
+        "github-cli": "latest"
     },
+    "mounts": [
+        "source=${env:HOME}/.ssh,target=/home/vscode/.ssh,type=bind,consistency=cached",
```
Comment (gcharest) [inline .devcontainer/devcontainer.json:17]: I need to comment out line 17 to be able to access my SSH configs from within the container.

**Review by patheard — APPROVED** [review]

**Review by gcharest — APPROVED** [review]
👍

**Pull request merged**